src/client/async_forward_backward.py

```diff
@@ -0,0 +1,215 @@
+import logging
+from typing import Optional, Union, List, Sequence, Tuple, Dict
+
+import torch
+from hivemind import DHT, P2P, get_logger, use_hivemind_log_handler
+from hivemind.moe.client.remote_expert_worker import RemoteExpertWorker
+from torch import nn
+
+import asyncio
+from src.client.sequence_manager import RemoteSequenceManager
+
+from hivemind import (
+    P2P,
+    get_logger,
+    nested_flatten,
+    serialize_torch_tensor,
+    use_hivemind_log_handler,
+)
+
+from hivemind.utils.nested import nested_compare, nested_flatten, nested_pack
+from hivemind.moe.client.expert import expert_forward, expert_backward
+from hivemind.moe.client.remote_expert_worker import RemoteExpertWorker
+from hivemind.p2p import StubBase
+
+from src.client.sequence_manager import RemoteSequenceManager
+from src.server.handler import TransformerConnectionHandler
+from src.data_structures import CHAIN_DELIMITER, RemoteSpanInfo, ModuleUID, RemoteSpanInfo, RPCInfo
+
+
+MAX_TOKENS_IN_BATCH=1024
+
+
+async def run_forward(
+    uid: ModuleUID, 
+    stub: StubBase,
+    rpc_info: RPCInfo,
+    *inputs: torch.Tensor,
+    **kwargs
+) -> Tuple[torch.Tensor, ...]:
```

Comment:
Nit: consider adding a single-line docstring OR rename run_forward into something specific (so that users can distinguish it from async_forward, expert_forward)